pkg/karmadactl/apply.go

```diff
@@ -0,0 +1,214 @@
+package karmadactl
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	restclient "k8s.io/client-go/rest"
+	kubectlapply "k8s.io/kubectl/pkg/cmd/apply"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/karmadactl/options"
+	"github.com/karmada-io/karmada/pkg/util/names"
+)
+
+var metadataAccessor = meta.NewAccessor()
+
+// CommandApplyOptions contains the input to the apply command.
+type CommandApplyOptions struct {
+	// global flags
+	options.GlobalCommandOptions
+	// apply flags
+	KubectlApplyFlags *kubectlapply.ApplyFlags
+	Namespace         string
+	AllClusters       bool
+
+	kubectlApplyOptions *kubectlapply.ApplyOptions
+}
+
+var (
+	applyLong = templates.LongDesc(i18n.T(`
+		Apply a configuration to a resource by file name or stdin and propagate them into member clusters.
+		The resource name must be specified. This resource will be created if it doesn't exist yet.
+		To use 'apply', always create the resource initially with either 'apply' or 'create --save-config'.
+
+		JSON and YAML formats are accepted.
+
+		Alpha Disclaimer: the --prune functionality is not yet complete. Do not use unless you are aware of what the current state is. See https://issues.k8s.io/34274.
+		
+		Note: It implements the function of 'kubectl apply' by default. 
+		If you want to propagate them into member clusters, please use 'kubectl apply --all-clusters'.`))
+
+	applyExample = templates.Examples(i18n.T(`
```

Comment:
It introduce `templates.Examples(i18n.T())`, I think it is good to improve format, we can update all example at #2129. let @RainbowMango to confirm.

Reply:
I have removed the dependency on `i18n` in this pr.

we can do it in #2129